DistributionSystemApi/MailLibrary/IMailService.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DistributionSystemApi.MailLibrary
+{
+    public interface IMailService<MailModel>
```

Comment:
I dont think that you need to use generics here, if you have any questions or concerns we can discuss it